astropy/nddata/nduncertainty.py

```diff
@@ -748,6 +782,13 @@ def _propagate_divide(self, other_uncert, result_data, correlation):
     def _data_unit_to_uncertainty_unit(self, value):
         return value
 
+    def _convert_to_variance(self):
+        return VarianceUncertainty(self.array ** 2, unit=self.unit ** 2)
+
+    @classmethod
+    def _convert_from_variance(cls, var_uncert):
+        return cls(var_uncert.array ** (1/2), unit=var_uncert.unit ** (1/2))
```

Comment:
Same issue as https://github.com/astropy/astropy/pull/12057/files#r854225811 when `self.unit is None`